frontend/src/hooks/useLastViewedProject.ts

```diff
@@ -9,9 +10,14 @@ export const useLastViewedProject = () => {
         return getLocalStorageItem(key);
     });
 
+    const { emitEvent } = useCustomEvent('lastViewedUpdated', () => {
+        setLastViewed(getLocalStorageItem(key));
+    });
+
     useEffect(() => {
         if (lastViewed) {
             setLocalStorageItem(key, lastViewed);
+            emitEvent();
```

Comment:
every time we update local storage in one hook, all other hooks created from the same fn will get their local state updated